### PR TITLE
Allow links only for social badges, affects [GitHubMilestone Dependabot]

### PR DESCRIPTION
**Pull request by PyvesB**
Related to #6096.

Twitch is also another badge that doesn't follow this, however, I'm suggesting #6158 as a resolution instead.

### CODE REVIEW

File: services/dependabot/dependabot.service.js [hunk 1]
```diff
@@ -58,7 +58,6 @@ module.exports = class DependabotSemverCompatibility extends BaseJsonService {
     return {
       color: json.colour,
       message: json.status,
-      link: this._getLink({ packageManager, dependencyName }),
```
Comment (chris48s) [inline services/dependabot/dependabot.service.js:61]: Do we still need to keep the `_getLink()` function itself? Seems like we could remove it..

Comment (PyvesB) [reply services/dependabot/dependabot.service.js:61]: Indeed, had missed it, good catch.

**Comment by chris48s** [comment]
sorry - one more request. Can we add a line to this section of docs

https://github.com/badges/shields/blob/master/CONTRIBUTING.md#badge-guidelines

saying default links are always off for non-social badges

**Review by chris48s — APPROVED** [review]

**Comment by PyvesB** [comment]
Deployed to production and communication on Twitter and Discord done.

**Pull request merged**